modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -318,11 +339,16 @@ object CalibrationsService extends CalibrationObservations {
                       )
                     )
                   )
-                )
               ),
               // Important: Only update the obs that need it or it will produce a cascade of infinite updates
               // TODO: This could be slightly optimized by grouping obs per configuration and updating in batches
-              sql"select $observation_id where c_calibration_role = $calibration_role and c_spec_signal_to_noise_at <> $wavelength_pm".apply(oid, role, w)
+              sql"""
+                SELECT $observation_id
+                  FROM t_observation
+                 WHERE c_observation_id = $observation_id
+                   AND c_calibration_role = $calibration_role
+                   AND ("""(oid, oid, role) |+| needsUpdate |+| void")"
+//              sql"select $observation_id where c_calibration_role = $calibration_role and c_spec_signal_to_noise_at <> $wavelength_pm".apply(oid, role, w)
```

Comment:
I reworked this query, but I may be misunderstanding the original intent so I'm particularly unsure about my update here.